ols/src/tools/oc_cli.py

```diff
@@ -0,0 +1,239 @@
+"""OpenShift CLI tools."""
+
+import logging
+import os
+import subprocess
+from typing import Optional
+
+from langchain.tools import tool
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: OLS-1443
+# TODO: might also sanitize the case when llm sends the full command instead
+# of just args, eg. [oc get pods -n bla] -> [pods -n bla]
+def sanitize_oc_args(args: list[str]) -> list[str]:
+    """Sanitize `oc` CLI arguments."""
+    blocked_chars = {";", "&", "|", "`", "$", "(", ")", "<", ">", "\\"}
+
+    sanitized_args = []
+    for arg in args:
+        safe_part = []
+        for char in arg:
+            if char in blocked_chars:
+                # stop processing further characters in this argument
+                logger.warning(
+                    "Problematic character(s) found in oc tool argument '%s'", arg
+                )
+                break
+            safe_part.append(char)
+
+        if safe_part:  # only add non-empty results
+            sanitized_args.append("".join(safe_part).strip())
+
+    return sanitized_args
+
+
+def run_oc(args: list[str]) -> subprocess.CompletedProcess:
+    """Run `oc` CLI with provided arguments and command."""
+    try:
+        res = subprocess.run(  # noqa: S603
+            ["oc", *args],  # noqa: S607
+            capture_output=True,
+            text=True,
+            check=True,
+            shell=False,
+        )
+        return res
+    except subprocess.CalledProcessError as e:
+        logger.error(
+            "Error running oc command %s: %s, stdout: %s, stderr: %s",
+            args,
+            e,
+            e.stdout,
+            e.stderr,
+        )
+        raise
+
+
+# TODO: server address configurable via request?
+def log_to_oc(
```

Comment:
@onmete Could you please explain why we need this login ? Why we cannot directly call the commands from model ?